<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Paige Ellestad and Sven Buerki - Boise State University" />

<meta name="date" content="2022-07-05" />

<title>GenomeScope review</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Vanilla genomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="reproducible_report.html">Reproducible report</a>
</li>
<li>
  <a href="heterozygosity_review.html">Heterozygosity</a>
</li>
<li>
  <a href="funding.html">Acknowledgements</a>
</li>
<li>
  <a href="citation.html">Citation</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GenomeScope review</h1>
<h3 class="subtitle">Vanilla genomics</h3>
<h4 class="author">Paige Ellestad and Sven Buerki - Boise State University</h4>
<h4 class="date">2022-07-05</h4>

</div>


<script type="text/javascript">
$(document).ready(function() {
  $('#header').parent().prepend('<div id=\"logo\"><img src=\"Images/boisestate-primarylogo-2color-rgb.png\" style=\"position:absolute; top:0; right:0; padding:10px; height:120px\"></div>');
  $('#header').css('margin-right', '120px')
});
</script>
<div style="text-align: right">
<a href="https://github.com/svenbuerki/VanillaGenomicsCode">Raw data on GitHub</a>
</div>
<!-- The website is here: https://svenbuerki.github.io/VanillaGenomicsCode/ -->
<div id="introduction" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p>To conduct a literature review on genomic heterozygosity levels in plants inferred using approaches implemented in GenomeScope <span class="citation">(<a href="#ref-genomescope" role="doc-biblioref">Vurture et al., 2017</a>)</span> and GenomeScope v.2.0 <span class="citation">(<a href="#ref-RanalloBenavidez2020" role="doc-biblioref">Ranallo-Benavidez et al., 2020</a>)</span>, an R function (<code>CitedInExtractoR</code>) was developed. This function takes unique PubMed IDs as input and retrieves citations of publications citing these studies. Publications were manually inspected to extract genomic heterozygosity values, taxonomy, ploidy and estimated genome size. Data were summarized using violin plots. Sections below provide the code to execute the analyses presented here.</p>
<div id="retriving-publications-citing-genomescope" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Retriving publications citing GenomeScope</h2>
<p>The function used to retrieve publications citing GenomeScope software is presented below and depends on <em>rentrez</em> and <em>easyPubMed</em> R packages. To install these packages do:</p>
<pre class="r"><code>#Install packages
install.packages(c(&quot;rentrez&quot;, &quot;easyPubMed&quot;))</code></pre>
<div id="citedinextractor-function" class="section level3" number="1.1.1">
<h3><span class="header-section-number">1.1.1</span> CitedInExtractoR function</h3>
<p>This function needs to be sourced prior to conducting the analyses.</p>
<pre class="r"><code>#A function to list publications citing a target article and 
# downloading data on these latter publications
CitedInExtractoR &lt;- function(PubMedID){
  print(paste(&quot;Fetch data associated to&quot;, PubMedID, sep = &quot; &quot;))
  #Search data associated to PubMedID in pubmed
  src &lt;- rentrez::entrez_link(dbfrom = &quot;pubmed&quot;, id = PubMedID, db = &quot;pubmed&quot;)
  
  print(paste(&quot;Extract list of publications citing&quot;, PubMedID, sep = &quot; &quot;))
  #Extract PubMedIDs of pubs citing our target Pub (= vect of pubmedIDs)
  citedIn &lt;- src$links$pubmed_pubmed_citedin
  print(paste(PubMedID, &quot;is cited in&quot;, length(citedIn), &quot;publications&quot;, sep = &quot; &quot;))
  #Create empty object to save output of loop
  OUT &lt;- NULL
  #Generate table with output data
  print(paste(&quot;Download data on publications citing&quot;, PubMedID, sep = &quot; &quot;))
  #Set a progress bar
  pb &lt;- txtProgressBar(min = 0, max = length(citedIn), style = 3)
  for(i in 1:length(citedIn)){
    RefData &lt;- rentrez::entrez_fetch(db = &quot;pubmed&quot;, id = citedIn[i], rettype = &quot;xml&quot;)
    tabPubMed &lt;- easyPubMed::article_to_df(RefData)[1,]
    OUT &lt;- rbind(OUT, tabPubMed)
    # update progress bar
    setTxtProgressBar(pb, i)
  }
  close(pb)
  
  #Add col with TargetPubMed ID
  TargetPubMed &lt;- rep(PubMedID, nrow(OUT))
  
  #Add col with clickable DOI
  paste(rep(&quot;http://doi.org/&quot;, nrow(OUT)), OUT$doi, sep = &quot;&quot;)
  #FINAL dataset
  FINAL &lt;- cbind(TargetPubMed, OUT)
  
  FINAL
}</code></pre>
</div>
<div id="applying-citedinextractor-to-retrive-publications" class="section level3" number="1.1.2">
<h3><span class="header-section-number">1.1.2</span> Applying CitedInExtractoR to retrive publications</h3>
<p><code>CitedInExtractoR</code> was applied on PubMed IDs for GenomScope publications to retrieve publications citing these software (executed on March 29th 2022).</p>
<pre class="r"><code># Load packages
library(rentrez)
library(easyPubMed)

# List PubMed IDs for GenomeScope publications
PubMedID &lt;- c(&quot;28369201&quot;, &quot;32188846&quot;)

# Apply function across PubMedId vector
DatPubMed &lt;- NULL
for(i in 1:length(PubMedID)){
  tmp &lt;- CitedInExtractoR(PubMedID[i])
  DatPubMed &lt;- rbind(DatPubMed, tmp)
  #Add a URLs to directly access to DOIs
  DatPubMed$URLDOI &lt;- paste(rep(&quot;http://doi.org/&quot;, nrow(DatPubMed)), DatPubMed$doi, sep = &quot;&quot;)
}

# Write data out (as csv)
write.csv(DatPubMed, file = &quot;PubCitedIn_GenomeScope_publications_28369201_32188846_29March2022.csv&quot;, row.names = F, quote=T)</code></pre>
</div>
</div>
<div id="summarizing-extracted-data" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Summarizing extracted data</h2>
<p>The code below was used to summarize the genomic data using the R package <em>ggplot2</em>.</p>
<pre class="r"><code>####Creating plots using genomic and plant usage data####
##Paige Ellestad
##June 6th, 2022

#load Genomescope review data
data &lt;- read.csv(&quot;PubCitedIn_GenomeScope_publications_28369201_32188846_29March2022 - PubCitedIn_GenomeScope_publications_28369201_32188846_29March2022.csv&quot;)

#change genome size and heterozygosity values to numericals
data$Heterozygosity_Perc &lt;- as.numeric(data$Heterozygosity_Perc)
data$GS_bp &lt;- as.numeric(data$GS_bp)

#For rows with ploidy levels as &quot;na&quot;, change to &quot;unknown&quot; (for plots)
data$Ploidy[which(data$Ploidy == &quot;na&quot;)] &lt;- &quot;unknown&quot;

#Make Ploidy levels and Plant assignation factors
data$Ploidy &lt;- as.factor(data$Ploidy)
data$Plant &lt;- as.factor(data$Plant)

##make two subsets of data for separate heterozygosity and genome size plots
#subset data to exclude rows without heterozygosity values
data_het &lt;- data[!is.na(data$Heterozygosity_Perc),]

#subset data to exclude rows without genome size values
data_gensize &lt;- data[!is.na(data$GS_bp),]

##use ggplot2 to make violin plots of distributions
library(ggplot2)

#heterozygosity plot
#assign labels to chosen species (this was manually done within the Genomescope review 
#dataset by adding a &quot;1&quot; in a new column &quot;figure&quot; for species to be labelled within the plot)
species_labels &lt;- data_het[which(data_het$figure == 1),]

#Make basic violin plot grouped by ploidy levels
grouped_plot &lt;- ggplot(data_het, aes(x=Ploidy, y=Heterozygosity_Perc)) + 
  geom_violin() 

#Make labelled, colored violin plot with all ploidy levels in one plot
het_plot &lt;- 
  ggplot(data_het, aes(x=Plant, y=Heterozygosity_Perc)) + 
  geom_violin() +
  geom_text(data=species_labels,aes(x=Plant, y = Heterozygosity_Perc,label=Species, 
                                    color = Ploidy),hjust=0, nudge_x = .01, vjust=.35, size= 3)+
  geom_point(aes(color = Ploidy)) +
  xlab(&quot;&quot;) +
  ylab(&quot;Genomic Heterozygosity(%)&quot;) +
  theme_classic() +
  theme(legend.title.align = 0.5, axis.ticks.x = element_blank(),
        axis.text.x = element_blank())

#Print to a pdf
pdf(&quot;het_violin_plot.pdf&quot;)
het_plot
dev.off()

#genome size plot
gensize_plot &lt;- 
  ggplot(data_gensize, aes(x=Plant, y=GS_bp)) + 
  geom_violin() +
  geom_point(aes(color = Ploidy)) + 
  xlab(&quot;&quot;) +
  ylab(&quot;Genome Size (bp)&quot;) +
  theme(axis.ticks.x = element_blank(),
        axis.text.x = element_blank())

#print to a pdf
pdf(&quot;gensize_violin_plot.pdf&quot;)
gensize_plot
dev.off()</code></pre>
</div>
<div id="output" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Output</h2>
<p>The graph showing variation in genomic heterozygosity levels by ploidy in plants inferred by GenomeScope is in Figure <a href="#fig:Hete">1.1</a> and data on genome size variation for the same data is presented in Figure <a href="#fig:GS">1.2</a>.</p>
<div class="figure"><span style="display:block;" id="fig:Hete"></span>
<img src="Images/het_violin_plot.png" alt="GenomeScope output showing variation in heterozygosity levels in plants." width="80%" />
<p class="caption">
Figure 1.1: GenomeScope output showing variation in heterozygosity levels in plants.
</p>
</div>
<div class="figure"><span style="display:block;" id="fig:GS"></span>
<img src="Images/gensize_violin_plot.png" alt="GenomeScope output showing variation in genome size levels in plants." width="80%" />
<p class="caption">
Figure 1.2: GenomeScope output showing variation in genome size levels in plants.
</p>
</div>
</div>
</div>
<div id="references" class="section level1" number="2">
<h1><span class="header-section-number">2</span> References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-RanalloBenavidez2020" class="csl-entry">
<span class="smallcaps">Ranallo-Benavidez, T.R.</span>, <span class="smallcaps">K.S. Jaron</span>, and <span class="smallcaps">M.C. Schatz</span>. 2020. GenomeScope 2.0 and smudgeplot for reference-free profiling of polyploid genomes. <em>Nature Communications</em> 11: 1432. Available at: <a href="https://doi.org/10.1038/s41467-020-14998-3">https://doi.org/10.1038/s41467-020-14998-3</a>.
</div>
<div id="ref-genomescope" class="csl-entry">
<span class="smallcaps">Vurture, G.W.</span>, <span class="smallcaps">F.J. Sedlazeck</span>, <span class="smallcaps">M. Nattestad</span>, <span class="smallcaps">C.J. Underwood</span>, <span class="smallcaps">H. Fang</span>, <span class="smallcaps">J. Gurtowski</span>, and <span class="smallcaps">M.C. Schatz</span>. 2017. GenomeScope: Fast reference-free genome profiling from short reads. <em>Bioinformatics</em> 33: 2202–2204. Available at:<a href=" + http://dx.doi.org/10.1093/bioinformatics/btx153"> + http://dx.doi.org/10.1093/bioinformatics/btx153</a>.
</div>
</div>
</div>
<div id="appendix-1" class="section level1 appendix" number="3">
<h1><span class="header-section-number">3</span> Appendix 1</h1>
<p>Citations of all R packages used to generate this report.</p>
<p>[1] J. Allaire, Y. Xie, J. McPherson, et al. <em>rmarkdown: Dynamic
Documents for R</em>. R package version 2.14. 2022.
<a href="https://CRAN.R-project.org/package=rmarkdown" class="uri">https://CRAN.R-project.org/package=rmarkdown</a>.</p>
<p>[2] C. Boettiger. <em>knitcitations: Citations for Knitr Markdown Files</em>.
R package version 1.0.12. 2021.
<a href="https://github.com/cboettig/knitcitations" class="uri">https://github.com/cboettig/knitcitations</a>.</p>
<p>[3] M. C. Koohafkan. <em>kfigr: Integrated Code Chunk Anchoring and
Referencing for R Markdown Documents</em>. R package version 1.2.1. 2021.
<a href="https://github.com/mkoohafkan/kfigr" class="uri">https://github.com/mkoohafkan/kfigr</a>.</p>
<p>[4] R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2022. <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>.</p>
<p>[5] H. Wickham, J. Bryan, and M. Barrett. <em>usethis: Automate Package
and Project Setup</em>. R package version 2.1.6. 2022.
<a href="https://CRAN.R-project.org/package=usethis" class="uri">https://CRAN.R-project.org/package=usethis</a>.</p>
<p>[6] H. Wickham, R. François, L. Henry, et al. <em>dplyr: A Grammar of Data
Manipulation</em>. R package version 1.0.9. 2022.
<a href="https://CRAN.R-project.org/package=dplyr" class="uri">https://CRAN.R-project.org/package=dplyr</a>.</p>
<p>[7] H. Wickham, J. Hester, W. Chang, et al. <em>devtools: Tools to Make
Developing R Packages Easier</em>. R package version 2.4.3. 2021.
<a href="https://CRAN.R-project.org/package=devtools" class="uri">https://CRAN.R-project.org/package=devtools</a>.</p>
<p>[8] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. ISBN 978-1138700109. Boca Raton, Florida: Chapman and
Hall/CRC, 2016. <a href="https://bookdown.org/yihui/bookdown" class="uri">https://bookdown.org/yihui/bookdown</a>.</p>
<p>[9] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. R package version 0.27. 2022.
<a href="https://CRAN.R-project.org/package=bookdown" class="uri">https://CRAN.R-project.org/package=bookdown</a>.</p>
<p>[10] Y. Xie. <em>Dynamic Documents with R and knitr</em>. 2nd. ISBN
978-1498716963. Boca Raton, Florida: Chapman and Hall/CRC, 2015.
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>.</p>
<p>[11] Y. Xie. <em>formatR: Format R Code Automatically</em>. R package version
1.12. 2022. <a href="https://github.com/yihui/formatR" class="uri">https://github.com/yihui/formatR</a>.</p>
<p>[12] Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in
R”. In: <em>Implementing Reproducible Computational Research</em>. Ed. by V.
Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and
Hall/CRC, 2014. <a href="http://www.crcpress.com/product/isbn/9781466561595" class="uri">http://www.crcpress.com/product/isbn/9781466561595</a>.</p>
<p>[13] Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.39. 2022.
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>.</p>
<p>[14] Y. Xie and J. Allaire. <em>tufte: Tufte’s Styles for R Markdown
Documents</em>. R package version 0.12. 2022.
<a href="https://github.com/rstudio/tufte" class="uri">https://github.com/rstudio/tufte</a>.</p>
<p>[15] Y. Xie, J. Allaire, and G. Grolemund. <em>R Markdown: The Definitive
Guide</em>. ISBN 9781138359338. Boca Raton, Florida: Chapman and Hall/CRC,
2018. <a href="https://bookdown.org/yihui/rmarkdown" class="uri">https://bookdown.org/yihui/rmarkdown</a>.</p>
<p>[16] Y. Xie, C. Dervieux, and E. Riederer. <em>R Markdown Cookbook</em>. ISBN
9780367563837. Boca Raton, Florida: Chapman and Hall/CRC, 2020.
<a href="https://bookdown.org/yihui/rmarkdown-cookbook" class="uri">https://bookdown.org/yihui/rmarkdown-cookbook</a>.</p>
<p>[17] H. Zhu. <em>kableExtra: Construct Complex Table with kable and Pipe
Syntax</em>. R package version 1.3.4. 2021.
<a href="https://CRAN.R-project.org/package=kableExtra" class="uri">https://CRAN.R-project.org/package=kableExtra</a>.</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
