<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sven Buerki and Paige Ellestad - Boise State University" />

<meta name="date" content="2022-05-24" />

<title>Analytical workflow and code associated to Vanilla genomics analyses</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Vanilla genomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Analytical workflow and code associated to <em>Vanilla</em> genomics analyses</h1>
<h4 class="author">Sven Buerki and Paige Ellestad - Boise State University</h4>
<h4 class="date">2022-05-24</h4>

</div>


<script type="text/javascript">
$(document).ready(function() {
  $('#header').parent().prepend('<div id=\"logo\"><img src=\"Images/boisestate-primarylogo-2color-rgb.png\" style=\"position:absolute; top:0; right:0; padding:10px; height:120px\"></div>');
  $('#header').css('margin-right', '120px')
});
</script>
<div style="text-align: right">
<a href="">Download pdf version</a>
</div>
<div style="text-align: right">
<a href="https://github.com/svenbuerki/VanillaGenomicsCode">Raw data on GitHub</a>
</div>
<div id="introduction" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p>This document provides the software, packages and code used to perform the genomic analyses described in Ellestad <em>et al.</em> (in prep.). All the analyses were performed on a Linux computer with the Ubuntu 18.04.6 LTS (GNU/Linux 4.15.0-177-generic x86_64) OS at Boise State University (ID, USA).</p>
</div>
<div id="data" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Data</h1>
<p>This project relies on pair-end Illumina short-read data (2x150bp) stored in <code>fastq</code> format and compressed using <code>gz</code> protocol. The data produced in this project will be deposited on NCBI under BioProject (pending). We are also using data from BioProject <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA633886/">PRJNA633886</a> published in <span class="citation"><a href="#ref-Hasing2020" role="doc-biblioref">Hasing et al.</a> (<a href="#ref-Hasing2020" role="doc-biblioref">2020</a>)</span>.</p>
</div>
<div id="project-structure" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Project structure</h1>
<p>Although the data underpinning the analyses are not presented here (due to size limitations), we are providing an overview of our project structure to facilitate understanding of the code (see Figure <a href="#fig:prjstr">3.1</a>).</p>
<div class="figure"><span style="display:block;" id="fig:prjstr"></span>
<img src="Images/Project_str_VanillaCode.png" alt="Overview of the project structure applied to analyze data for the Vanilla genomics project." width="100%" />
<p class="caption">
Figure 3.1: Overview of the project structure applied to analyze data for the Vanilla genomics project.
</p>
</div>
<p>A summary of the content of each folder is provided here (in alphabetical order):</p>
<ul>
<li><code>Bowtie</code>: Contains <code>sam</code>, <code>bam</code>, <code>bcf</code> and <code>fa</code> files associated to read mapping analyses to assemble <em>Vanilla</em> genomes from Illumina short-read data using the <em>Vanilla planifolia</em> (Daphna) reference genome (<a href="https://www.ncbi.nlm.nih.gov/data-hub/genome/GCA_016413895.1/">GCA_016413895.1</a>).</li>
<li><code>DotPlot</code>: Contains <code>html</code> and <code>png</code> files associated to the dotplot analyses of the <em>Vanilla</em> genomes.</li>
<li><code>GenomeScope</code>: Contains GenomeScope output files (<code>png</code>) generated using output of the JellyFish analyses.</li>
<li><code>Heatmap</code>: Contains <code>csv</code> and <code>pdf</code> files associated to the percentage identity analyses at chromosome level.</li>
<li><code>JellyFish</code>: Contains the <code>histo</code> files associated to the k-mer frequency analyses conducted in JellyFish and used as input for the <code>GenomeScope</code> analyses.</li>
<li><code>Minimap2</code>: Contains the <code>paf</code> files from the <code>Minimap2</code> analyses used to conduct the dotplot, heatmap and genome coverage analyses.</li>
<li><code>Pafr</code>: Contains <code>pdf</code> files associated to the genome coverage maps inferred using <code>paf</code> files. This folder is entitled <code>Pafr</code> because the coverage maps are inferred using the r package <em>pafr</em>.</li>
<li><code>Raw_Data</code>: Contains <code>fastq.gz</code> files associated to the raw Illumina data used for this project.</li>
<li><code>Reference_genome</code>: Contains <code>fna</code> files for the <em>Vanilla planifolia</em> (Daphna) reference genome (<a href="https://www.ncbi.nlm.nih.gov/data-hub/genome/GCA_016413895.1/">GCA_016413895.1</a>) as well as the index files for the read mapping analyses.</li>
<li><code>Smudgeplot</code>: Contains <code>kmctools</code> output files used for the Smudgeplot analyses.</li>
<li><code>Trimmomatic</code>: Contains <code>trimmed.fastq.gz</code> files used as input for the genomic analyses.</li>
</ul>
</div>
<div id="analytical-pipeline" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Analytical pipeline</h1>
<p>This section provides the code associated with the analyses aiming at comparing the genome complexity and structure of samples of <em>Vanilla planifolia</em> from Mexico and data available on NCBI. The following tasks are covered:</p>
<ol style="list-style-type: decimal">
<li>Setting environment.</li>
<li>Downloading SRA files.</li>
<li>Conducting reads trimming.</li>
<li>Inferring genome size and heterozygosity.</li>
<li>Inferring ploidy and genome complexity.</li>
<li>Reconstructing genome sequences.</li>
<li>Testing for chromosomal structural re-arrangements and inferring percentage identities.</li>
</ol>
<div id="setting-environment" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Setting environment</h2>
<p>Since these analyses are requiring a UNIX-based OS, we started by <em>i</em>) remotely accessing the Linux computer using <code>ssh</code> protocol (you need your user ID and IP address), <em>ii</em>) launching a <code>tmux</code> session (safe environment that can be accessed from different locations) and <em>iii</em>) navigating to project folder (here <code>Vanilla_SRA</code>) and <em>iv</em>) creating a variable with sample ID for analyses as follows:</p>
<pre class="bash"><code># 1. Remotely connect to computer using ssh (adjust with your credentials/IP)
ssh userID@IP

# 2. Start tmux session (to exit tmux session type Ctrl+b and d)
tmux

# 3.  Navigate to project location (adjust path)
cd /media/SeaGate/Vanilla_SRA/Raw_Data/</code></pre>
<p>Further information on <code>ssh</code> and <code>tmux</code> protocols can be accessed <a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Tutorials.html">here</a>.</p>
</div>
<div id="downloading-sra-files" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Downloading SRA files</h2>
<p>Here, we need to <em>i</em>) create the <code>$SRA</code> variable with <code>SRA</code> accession number of data to be downloaded, <em>ii</em>) navigate to right folder (where data will be saved; here <code>Raw_Data/</code>) and <em>iii</em>) download <code>SRA</code> files (R1 and R2) using <code>parallel-fastq-dump</code>. This program allows parallelizing the downloading process and greatly reduces the downloading time. In this case, the files that we are downloading are compressed (<code>gz</code>) and they are in <code>fastq</code> format generated by Illumina instruments. The commands are available below:</p>
<pre class="bash"><code># 1. Create variable with sample SRA accession (as provided on NCBI)
SRA=SRR12628846 &amp;&amp; # V. planifolia &#39;Hawaii&#39;

# 2. Navigate to right location
cd Raw_Data/ &amp;&amp;

# 3. Download SRA files (R1 and R2) with parallelfastq (using 26 threads)
parallel-fastq-dump -O . --sra-id $SRA --threads 26 --split-files --gzip</code></pre>
<div id="install-parallel-fastq-dump" class="section level3" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Install <code>parallel-fastq-dump</code></h3>
<p><code>parallel-fastq-dump</code> is available on <a href="https://github.com/rvalieris/parallel-fastq-dump">GitHub</a> and can be installed as follows:</p>
<pre class="bash"><code># Install program with Bioconda
conda install parallel-fastq-dump</code></pre>
</div>
</div>
<div id="conducting-reads-trimming" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Conducting reads trimming</h2>
<p>Here, we use <code>Trimmomatic</code> <span class="citation">(<a href="#ref-Bolger2014" role="doc-biblioref">Bolger et al., 2014</a>)</span> to clean/trim the Illumina reads as follows:</p>
<pre class="bash"><code># 1. Navigate to right location
cd ../Trimmomatic &amp;&amp;

# 2. Conduct reads trimming
TrimmomaticPE -threads 30 /media/SeaGate/Vanilla_SRA/Raw_Data/&quot;$SRA&quot;_1.fastq.gz /media/SeaGate/Vanilla_SRA/Raw_Data/&quot;$SRA&quot;_2.fastq.gz &quot;$SRA&quot;_1_trimmed.fastq.gz &quot;$SRA&quot;_1_UNPAIRED.fastq.gz &quot;$SRA&quot;_2_trimmed.fastq.gz &quot;$SRA&quot;_2_UNPAIRED.fastq.gz LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:100</code></pre>
<div id="install-trimmomatic" class="section level3" number="4.3.1">
<h3><span class="header-section-number">4.3.1</span> Install <code>Trimmomatic</code></h3>
<p><code>Trimmomatic</code> is available on <a href="https://github.com/usadellab/Trimmomatic">GitHub</a>.</p>
</div>
</div>
<div id="inferring-genome-size-and-heterozygosity" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Inferring genome size and heterozygosity</h2>
<p>To infer the genome size and level of heterozygosity of <em>Vanilla planifolia</em> samples based on cleaned trimmed data, we are using the method implemented in <code>GenomScope 2.0</code> <span class="citation">(<a href="#ref-RanalloBenavidez2020" role="doc-biblioref">Ranallo-Benavidez et al., 2020</a>)</span>, which relies on k-mers (k=21) inferred by <code>JellyFish</code> <span class="citation">(<a href="#ref-Jellyfish2011" role="doc-biblioref">Marçais and Kingsford, 2011</a>)</span>. The commands are as follows:</p>
<pre class="bash"><code># 1. Navigate to right location
cd ../JellyFish &amp;&amp;

# 2. Generate histo file with JellyFish
jellyfish count -t 30 -C -m 21 -s 5G -o 21mer_trimmed_&quot;$SRA&quot;_pe12 &lt;(zcat /media/SeaGate/Vanilla_SRA/Trimmomatic/&quot;$SRA&quot;_1_trimmed.fastq.gz) &lt;(zcat /media/SeaGate/Vanilla_SRA/Trimmomatic/&quot;$SRA&quot;_2_trimmed.fastq.gz) &amp;&amp; 
jellyfish histo -o 21mer_trimmed_&quot;$SRA&quot;_pe12.histo 21mer_trimmed_&quot;$SRA&quot;_pe12

# 3. Use GenomeScope 2.0 to infer genome size and level of heterozygosity
# Run locally using GenomeScope R package (must be installed)
genomescope.R -i 21mer_trimmed_&quot;$SRA&quot;_pe12.histo -o output -k 21
# Or go online ti run analysis with histo file
http://qb.cshl.edu/genomescope/genomescope2.0/</code></pre>
<p><strong>Note:</strong> You need to record <code>kcov</code> from the <code>GenomeScope</code> output to set the <code>L</code> parameter in the <code>smudgeplot</code> analysis (see below).</p>
<div id="install-software" class="section level3" number="4.4.1">
<h3><span class="header-section-number">4.4.1</span> Install software</h3>
<p><code>GenomScope 2.0</code> is available on <a href="https://github.com/tbenavi1/genomescope2.0">GitHub</a> and <code>JellyFish</code> is available <a href="https://genome.umd.edu/jellyfish.html#Release">here</a>.</p>
</div>
</div>
<div id="inferring-ploidy-and-genome-complexity." class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> Inferring ploidy and genome complexity.</h2>
<p>We have inferred ploidy level of <em>Vanilla planifolia</em> samples and further investigated their genomic complexity (focusing on heterozygous k-mers) by applying the approach implemented in <code>smudgeplot</code> <span class="citation">(<a href="#ref-RanalloBenavidez2020" role="doc-biblioref">Ranallo-Benavidez et al., 2020</a>)</span>, which relies on k-mers (k=21) inferred by <code>KMC</code> <span class="citation">(<a href="#ref-KMC2017" role="doc-biblioref">Kokot et al., 2017</a>)</span>. The commands are as follows:</p>
<pre class="bash"><code># 1.Navigate to right folder
cd ../Smudgeplot/ &amp;&amp;

# 2. Conduct smudgeplot to infer ploidy and genome complexity
# list files for kmc tools analysis
ls /media/SeaGate/Vanilla_SRA/Trimmomatic/&quot;$SRA&quot;_R*_trimmed.fq.gz &gt; FILES_&quot;$SRA&quot; &amp;&amp;
# kmer 21, 30 threads, 128G of memory, counting kmer coverages between 1 and 10000x
kmc -k21 -t30 -m128 -ci1 -cs10000 @FILES_&quot;$SRA&quot; kmcdb_&quot;$SRA&quot; . &amp;&amp;
kmc_tools transform kmcdb_&quot;$SRA&quot; histogram kmcdb_k21_&quot;$SRA&quot;.hist -cx10000 &amp;&amp;

# Declare variable 
kcov=32 #Please edit based on GenomeScope analysis

# Define L = (kcov/2) - 5 # As proposed in smudgeplot manual
L=$(expr $kcov / 2 - 5) &amp;&amp;
U=$(smudgeplot.py cutoff kmcdb_k21_&quot;$SRA&quot;.hist U) &amp;&amp;

echo $L $U # these need to be sane values &amp;&amp;
# L should be like 20 - 200
# U should be like 500 - 3000

kmc_tools transform kmcdb_&quot;$SRA&quot; -ci&quot;$L&quot; -cx&quot;$U&quot; reduce kmcdb_&quot;$SRA&quot;_L&quot;$L&quot;_U&quot;$U&quot; &amp;&amp;
smudge_pairs kmcdb_&quot;$SRA&quot;_L&quot;$L&quot;_U&quot;$U&quot; kmcdb_&quot;$SRA&quot;_L&quot;$L&quot;_U&quot;$U&quot;_coverages.tsv kmcdb_&quot;$SRA&quot;_L&quot;$L&quot;_U&quot;$U&quot;_pairs.tsv &gt; kmcdb_&quot;$SRA&quot;_L&quot;$L&quot;_U&quot;$U&quot;_familysizes.tsv

# Generate the smudgeplot
#Run R code to produce smudgeplot (R package must be installed)
smudgeplot.py plot kmcdb_&quot;$SRA&quot;_L&quot;$L&quot;_U&quot;$U&quot;_coverages.tsv -o &quot;$SRA&quot;_L&quot;$L&quot;_U&quot;$U&quot; -k 21</code></pre>
<div id="install-software-1" class="section level3" number="4.5.1">
<h3><span class="header-section-number">4.5.1</span> Install software</h3>
<p><code>KMC</code> is available on <a href="https://github.com/refresh-bio/KMC">GitHub</a> and <code>smudgeplot</code> is also available on <a href="https://github.com/KamilSJaron/smudgeplot">GitHub</a>.</p>
</div>
</div>
</div>
<div id="references" class="section level1" number="5">
<h1><span class="header-section-number">5</span> References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Bolger2014" class="csl-entry">
<span class="smallcaps">Bolger, A.M.</span>, <span class="smallcaps">M. Lohse</span>, and <span class="smallcaps">B. Usadel</span>. 2014. Trimmomatic: A flexible trimmer for illumina sequence data. <em>Bioinformatics</em> 30: 2114–2120. Available at: <a href="http://dx.doi.org/10.1093/bioinformatics/btu170">http://dx.doi.org/10.1093/bioinformatics/btu170</a>.
</div>
<div id="ref-Hasing2020" class="csl-entry">
<span class="smallcaps">Hasing, T.</span>, <span class="smallcaps">H. Tang</span>, <span class="smallcaps">M. Brym</span>, <span class="smallcaps">F. Khazi</span>, <span class="smallcaps">T. Huang</span>, and <span class="smallcaps">A.H. Chambers</span>. 2020. A phased vanilla planifolia genome enables genetic improvement of flavour and production. <em>Nature Food</em> 1: 811–819. Available at: <a href="https://doi.org/10.1038/s43016-020-00197-2">https://doi.org/10.1038/s43016-020-00197-2</a>.
</div>
<div id="ref-KMC2017" class="csl-entry">
<span class="smallcaps">Kokot, M.</span>, <span class="smallcaps">M. Długosz</span>, and <span class="smallcaps">S. Deorowicz</span>. 2017. <span class="nocase">KMC 3: counting and manipulating k-mer statistics</span>. <em>Bioinformatics</em> 33: 2759–2761. Available at: <a href="https://doi.org/10.1093/bioinformatics/btx304">https://doi.org/10.1093/bioinformatics/btx304</a>.
</div>
<div id="ref-Jellyfish2011" class="csl-entry">
<span class="smallcaps">Marçais, G.</span>, and <span class="smallcaps">C. Kingsford</span>. 2011. A fast, lock-free approach for efficient parallel counting of occurrences of k-mers. <em>Bioinformatics</em> 27: 764–770. Available at: <a href="+ http://dx.doi.org/10.1093/bioinformatics/btr011">+ http://dx.doi.org/10.1093/bioinformatics/btr011</a>.
</div>
<div id="ref-RanalloBenavidez2020" class="csl-entry">
<span class="smallcaps">Ranallo-Benavidez, T.R.</span>, <span class="smallcaps">K.S. Jaron</span>, and <span class="smallcaps">M.C. Schatz</span>. 2020. GenomeScope 2.0 and smudgeplot for reference-free profiling of polyploid genomes. <em>Nature Communications</em> 11: 1432. Available at: <a href="https://doi.org/10.1038/s41467-020-14998-3">https://doi.org/10.1038/s41467-020-14998-3</a>.
</div>
</div>
</div>
<div id="appendix-1" class="section level1 appendix" number="6">
<h1><span class="header-section-number">6</span> Appendix 1</h1>
<p>Citations of all R packages used to generate this report.</p>
<p>[1] J. Allaire, Y. Xie, J. McPherson, et al. <em>rmarkdown: Dynamic
Documents for R</em>. R package version 2.14. 2022. &lt;URL:
<a href="https://CRAN.R-project.org/package=rmarkdown" class="uri">https://CRAN.R-project.org/package=rmarkdown</a>&gt;.</p>
<p>[2] C. Boettiger. <em>knitcitations: Citations for Knitr Markdown Files</em>.
R package version 1.0.12. 2021. &lt;URL:
<a href="https://github.com/cboettig/knitcitations" class="uri">https://github.com/cboettig/knitcitations</a>&gt;.</p>
<p>[3] M. C. Koohafkan. <em>kfigr: Integrated Code Chunk Anchoring and
Referencing for R Markdown Documents</em>. R package version 1.2.1. 2021.
&lt;URL: <a href="https://github.com/mkoohafkan/kfigr" class="uri">https://github.com/mkoohafkan/kfigr</a>&gt;.</p>
<p>[4] R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2019. &lt;URL: <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>&gt;.</p>
<p>[5] H. Wickham, J. Bryan, and M. Barrett. <em>usethis: Automate Package
and Project Setup</em>. R package version 2.1.5. 2021. &lt;URL:
<a href="https://CRAN.R-project.org/package=usethis" class="uri">https://CRAN.R-project.org/package=usethis</a>&gt;.</p>
<p>[6] H. Wickham, R. François, L. Henry, et al. <em>dplyr: A Grammar of Data
Manipulation</em>. R package version 1.0.9. 2022. &lt;URL:
<a href="https://CRAN.R-project.org/package=dplyr" class="uri">https://CRAN.R-project.org/package=dplyr</a>&gt;.</p>
<p>[7] H. Wickham, J. Hester, W. Chang, et al. <em>devtools: Tools to Make
Developing R Packages Easier</em>. R package version 2.4.3. 2021. &lt;URL:
<a href="https://CRAN.R-project.org/package=devtools" class="uri">https://CRAN.R-project.org/package=devtools</a>&gt;.</p>
<p>[8] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. ISBN 978-1138700109. Boca Raton, Florida: Chapman and
Hall/CRC, 2016. &lt;URL: <a href="https://bookdown.org/yihui/bookdown" class="uri">https://bookdown.org/yihui/bookdown</a>&gt;.</p>
<p>[9] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. R package version 0.26. 2022. &lt;URL:
<a href="https://CRAN.R-project.org/package=bookdown" class="uri">https://CRAN.R-project.org/package=bookdown</a>&gt;.</p>
<p>[10] Y. Xie. <em>Dynamic Documents with R and knitr</em>. 2nd. ISBN
978-1498716963. Boca Raton, Florida: Chapman and Hall/CRC, 2015. &lt;URL:
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>&gt;.</p>
<p>[11] Y. Xie. <em>formatR: Format R Code Automatically</em>. R package version
1.12. 2022. &lt;URL: <a href="https://github.com/yihui/formatR" class="uri">https://github.com/yihui/formatR</a>&gt;.</p>
<p>[12] Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in
R.” In: <em>Implementing Reproducible Computational Research</em>. Ed. by V.
Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and
Hall/CRC, 2014. &lt;URL:
<a href="http://www.crcpress.com/product/isbn/9781466561595" class="uri">http://www.crcpress.com/product/isbn/9781466561595</a>&gt;.</p>
<p>[13] Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.39. 2022. &lt;URL:
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>&gt;.</p>
<p>[14] Y. Xie and J. Allaire. <em>tufte: Tufte’s Styles for R Markdown
Documents</em>. R package version 0.12. 2022. &lt;URL:
<a href="https://github.com/rstudio/tufte" class="uri">https://github.com/rstudio/tufte</a>&gt;.</p>
<p>[15] Y. Xie, J. Allaire, and G. Grolemund. <em>R Markdown: The Definitive
Guide</em>. ISBN 9781138359338. Boca Raton, Florida: Chapman and Hall/CRC,
2018. &lt;URL: <a href="https://bookdown.org/yihui/rmarkdown" class="uri">https://bookdown.org/yihui/rmarkdown</a>&gt;.</p>
<p>[16] Y. Xie, C. Dervieux, and E. Riederer. <em>R Markdown Cookbook</em>. ISBN
9780367563837. Boca Raton, Florida: Chapman and Hall/CRC, 2020. &lt;URL:
<a href="https://bookdown.org/yihui/rmarkdown-cookbook" class="uri">https://bookdown.org/yihui/rmarkdown-cookbook</a>&gt;.</p>
<p>[17] H. Zhu. <em>kableExtra: Construct Complex Table with kable and Pipe
Syntax</em>. R package version 1.3.4. 2021. &lt;URL:
<a href="https://CRAN.R-project.org/package=kableExtra" class="uri">https://CRAN.R-project.org/package=kableExtra</a>&gt;.</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
