<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Paige Ellestad and Sven Buerki - Boise State University" />

<meta name="date" content="2022-05-27" />

<title>Reproducible report</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.23/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<script src="site_libs/jszip-1.11.3/jszip.min.js"></script>
<script src="site_libs/pdfmake-1.11.3/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.11.3/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.11.3/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.11.3/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Vanilla genomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="reproducible_report.html">Reproducible report</a>
</li>
<li>
  <a href="heterozygosity_review.html">Heterozygosity</a>
</li>
<li>
  <a href="location_pop.html">Map</a>
</li>
<li>
  <a href="funding.html">Acknowledgements</a>
</li>
<li>
  <a href="citation.html">Citation</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Reproducible report</h1>
<h3 class="subtitle">Vanilla genomics</h3>
<h4 class="author">Paige Ellestad and Sven Buerki - Boise State University</h4>
<h4 class="date">2022-05-27</h4>

</div>


<script type="text/javascript">
$(document).ready(function() {
  $('#header').parent().prepend('<div id=\"logo\"><img src=\"Images/boisestate-primarylogo-2color-rgb.png\" style=\"position:absolute; top:0; right:0; padding:10px; height:120px\"></div>');
  $('#header').css('margin-right', '120px')
});
</script>
<div style="text-align: right">
<a href="https://github.com/svenbuerki/VanillaGenomicsCode">Raw data on GitHub</a>
</div>
<!-- The website is here: https://svenbuerki.github.io/VanillaGenomicsCode/ -->
<div id="introduction" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p>This reproducible report provides evidence on the data, software, packages and code used to perform the genomic analyses described in Ellestad <em>et al.</em> (in prep.). All the analyses were performed on a DELL Precision 7920 with the Ubuntu 18.04.6 LTS (GNU/Linux 4.15.0-177-generic x86_64) OS at Boise State University (ID, USA) between January and June 2022.</p>
</div>
<div id="data" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Data</h1>
<p>This project relies on paired-end Illumina short-read data (2x150bp) in <code>fastq.gz</code> format generated on an Illumina HiSeq 2500 instrument. The data produced in this project are deposited on NCBI under BioProject (PRJNA841950) and SRA accessions SRR19374404-SRR19374419. We are also using data from the BioProject <a href="https://www.ncbi.nlm.nih.gov/bioproject/PRJNA633886/">PRJNA633886</a> published in <span class="citation"><a href="#ref-Hasing2020" role="doc-biblioref">Hasing et al.</a> (<a href="#ref-Hasing2020" role="doc-biblioref">2020</a>)</span>. Please find below further information on our sampling (herbarium specimens are deposited at HEM, Universidad de Ciencias y Artes de Chiapas, Mexico):</p>
<div id="htmlwidget-2cfd242493f95794a1f2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-2cfd242493f95794a1f2">{"x":{"filter":"none","vertical":false,"extensions":["Buttons"],"data":[["MEX12","MEX13","MEX14","MEX19","MEX20","MEX26","MEX31","MEX36","MEX41","MEX51","MEX59","MEX67","MEX69","MEX79"],["SAMN28632720","SAMN28632721","SAMN28632722","SAMN28632723","SAMN28632724","SAMN28632725","SAMN28632726","SAMN28632727","SAMN28632728","SAMN28632729","SAMN28632730","SAMN28632732","SAMN28632733","SAMN28632734"],["SRR19374418","SRR19374411","SRR19374410","SRR19374409","SRR19374408","SRR19374407","SRR19374406","SRR19374405","SRR19374404","SRR19374417","SRR19374416","SRR19374414","SRR19374413","SRR19374412"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Sample_ID<\/th>\n      <th>BioSample<\/th>\n      <th>SRA<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Blfrtip","buttons":["copy","csv","excel","pdf","print"],"columnDefs":[],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[0]; $(this.api().cell(row, 0).node()).css({'font-weight':'bold'});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
<div id="prjstr" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Project structure</h1>
<p>Although the data underpinning the analyses are not presented here (due to size limitations), we are providing an overview of our project structure to facilitate understanding of the code (see Figure <a href="#fig:prjstr">3.1</a>).</p>
<div class="figure"><span style="display:block;" id="fig:prjstr"></span>
<img src="Images/Project_str_VanillaCode.png" alt="Overview of the project structure applied to analyze data for the Vanilla genomics project." width="100%" />
<p class="caption">
Figure 3.1: Overview of the project structure applied to analyze data for the Vanilla genomics project.
</p>
</div>
<p>A summary of the content of each folder is provided here (in alphabetical order):</p>
<ul>
<li><code>Bowtie</code>: Contains <code>sam</code>, <code>bam</code>, <code>bcf</code> and <code>fa</code> files associated to read mapping analyses to assemble <em>Vanilla</em> genomes from Illumina short-read data using the <em>Vanilla planifolia</em> (Daphna) reference genome (<a href="https://www.ncbi.nlm.nih.gov/data-hub/genome/GCA_016413895.1/">GCA_016413895.1</a>).</li>
<li><code>DotPlot</code>: Contains <code>html</code> and <code>png</code> files associated to the dotplot analyses of the <em>Vanilla</em> genomes.</li>
<li><code>GenomeScope</code>: Contains GenomeScope output files (<code>png</code>) generated using output of the JellyFish analyses.</li>
<li><code>Heatmap</code>: Contains <code>csv</code> and <code>pdf</code> files associated to the percentage identity analyses at chromosome level.</li>
<li><code>JellyFish</code>: Contains the <code>histo</code> files associated to the k-mer frequency analyses conducted in JellyFish and used as input for the <code>GenomeScope</code> analyses.</li>
<li><code>Minimap2</code>: Contains the <code>paf</code> files from the <code>Minimap2</code> analyses used to conduct the dotplot, heatmap and genome coverage analyses.</li>
<li><code>Pafr</code>: Contains <code>pdf</code> files associated to the genome coverage maps inferred using <code>paf</code> files. This folder is entitled <code>Pafr</code> because the coverage maps are inferred using the r package <em>pafr</em>.</li>
<li><code>Raw_Data</code>: Contains <code>fastq.gz</code> files associated to the raw Illumina data used for this project.</li>
<li><code>Reference_genome</code>: Contains <code>fna</code> files for the <em>Vanilla planifolia</em> (Daphna) reference genome (<a href="https://www.ncbi.nlm.nih.gov/data-hub/genome/GCA_016413895.1/">GCA_016413895.1</a>) as well as the index files for the read mapping analyses.</li>
<li><code>Smudgeplot</code>: Contains <code>kmctools</code> output files used for the Smudgeplot analyses.</li>
<li><code>Trimmomatic</code>: Contains <code>trimmed.fastq.gz</code> files used as input for the genomic analyses.</li>
</ul>
</div>
<div id="overview-of-analytical-workflow" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Overview of analytical workflow</h1>
<p>This section provides an overview of the analyses conducted in this study aiming at comparing the genome complexity and structure of samples of <em>Vanilla planifolia</em> from Mexico and data available on NCBI. Detials on the software and associated code for the following analyses are presented in this document:</p>
<ol style="list-style-type: decimal">
<li>Setting environment.</li>
<li>Downloading SRA files.</li>
<li>Conducting reads trimming.</li>
<li>Genome size and heterozygosity.</li>
<li>Ploidy level and genome complexity.</li>
<li>Reconstructing genome sequences.</li>
<li>Comparative chromosome level analyses.</li>
</ol>
</div>
<div id="setting-environment" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Setting environment</h1>
<div id="dependencies" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Dependencies</h2>
<p>Analyses presented in this report are requiring a UNIX-based OS (tested on Ubuntu 18.04.6 LTS) and <code>ssh</code> and <code>tmux</code> protocols should be installed. Please see this <a href="https://svenbuerki.github.io/Genomics-Bioinformatics/Tutorials.html">website</a> for more details and documentation.</p>
</div>
<div id="code" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Code</h2>
<p>We start by <em>i</em>) remotely accessing the Linux computer using <code>ssh</code> protocol (you need your user ID and IP address), <em>ii</em>) launching a <code>tmux</code> session (safe environment that can be accessed from different locations) and <em>iii</em>) navigating to the project folder (here <code>Vanilla_SRA</code>) as follows:</p>
<pre class="bash"><code># 1. Remotely connect to computer using ssh (adjust with your credentials/IP)
ssh userID@IP

# 2. Start tmux session (to exit tmux session type Ctrl+b and d)
tmux

# 3.  Navigate to project location (adjust path)
cd /media/SeaGate/Vanilla_SRA/Raw_Data/</code></pre>
<p>More information on the project structure can be found <a href="#prjstr">here</a>.</p>
</div>
</div>
<div id="downloading-sra-files" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Downloading SRA files</h1>
<div id="data-requirements" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Data requirements</h2>
<ul>
<li><strong>Input:</strong> <code>SRA</code> accession number from NCBI.</li>
<li><strong>Output:</strong> Split <code>fastq.gz</code> files with R1 and R2 paired-end reads.</li>
</ul>
</div>
<div id="dependencies-1" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Dependencies</h2>
<p><code>SRA</code> files are downloaded using <code>parallel-fastq-dump</code>, which is available on <a href="https://github.com/rvalieris/parallel-fastq-dump">GitHub</a>. It can also be installed with <a href="https://anaconda.org/bioconda">Bioconda</a> as follows:</p>
<pre class="bash"><code># Install program with Bioconda
conda install parallel-fastq-dump</code></pre>
</div>
<div id="code-1" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> Code</h2>
<p>Here, we need to <em>i</em>) create the <code>$SRA</code> variable with <code>SRA</code> accession number of data to be downloaded, <em>ii</em>) navigate to right folder (where data will be saved; here <code>Raw_Data/</code>) and <em>iii</em>) download <code>SRA</code> files (R1 and R2) using <code>parallel-fastq-dump</code>. This program allows parallelizing the downloading process and greatly reduces the downloading time. In this case, the files that we are downloading are compressed (<code>gz</code>) and they are in <code>fastq</code> format generated by Illumina instruments. The commands are available below:</p>
<pre class="bash"><code># 1. Create variable with sample SRA accession (as provided on NCBI)
SRA=SRR12628846 &amp;&amp; # V. planifolia &#39;Hawaii&#39;

# 2. Navigate to right location
cd Raw_Data/ &amp;&amp;

# 3. Download SRA files (R1 and R2) with parallelfastq (using 26 threads)
parallel-fastq-dump -O . --sra-id $SRA --threads 26 --split-files --gzip</code></pre>
</div>
</div>
<div id="conducting-reads-trimming" class="section level1" number="7">
<h1><span class="header-section-number">7</span> Conducting reads trimming</h1>
<p>Here, we use <code>Trimmomatic</code> <span class="citation">(<a href="#ref-Bolger2014" role="doc-biblioref">Bolger et al., 2014</a>)</span> to clean/trim the Illumina paired-end reads from previous step.</p>
<div id="data-requirements-1" class="section level2" number="7.1">
<h2><span class="header-section-number">7.1</span> Data requirements</h2>
<ul>
<li><strong>Input:</strong> For each <code>SRA</code>, R1 and R2 <code>fastq.gz</code> files.</li>
<li><strong>Output:</strong> Four files, but only <code>$SRA_R1_trimmed.fastq.gz</code> and <code>$SRA_R2_trimmed.fastq.gz</code> containing cleaned paired reads from <code>Trimmomatic</code> will be used for downstream analyses.</li>
</ul>
</div>
<div id="dependencies-2" class="section level2" number="7.2">
<h2><span class="header-section-number">7.2</span> Dependencies</h2>
<p>You can obtain and install <code>Trimmomatic</code> directly from <a href="https://github.com/usadellab/Trimmomatic">GitHub</a>.</p>
</div>
<div id="code-2" class="section level2" number="7.3">
<h2><span class="header-section-number">7.3</span> Code</h2>
<p>To clean/trim the Illumina reads do as follows:</p>
<pre class="bash"><code># 1. Navigate to right location
cd ../Trimmomatic &amp;&amp;

# 2. Conduct reads trimming
TrimmomaticPE -threads 30 /media/SeaGate/Vanilla_SRA/Raw_Data/&quot;$SRA&quot;_1.fastq.gz /media/SeaGate/Vanilla_SRA/Raw_Data/&quot;$SRA&quot;_2.fastq.gz &quot;$SRA&quot;_1_trimmed.fastq.gz &quot;$SRA&quot;_1_UNPAIRED.fastq.gz &quot;$SRA&quot;_2_trimmed.fastq.gz &quot;$SRA&quot;_2_UNPAIRED.fastq.gz LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:100</code></pre>
</div>
</div>
<div id="genome-size-and-heterozygosity" class="section level1" number="8">
<h1><span class="header-section-number">8</span> Genome size and heterozygosity</h1>
<p>In this section, we are providing information to estimate genome sizes and levels of heterozygosity for the <em>Vanilla planifolia</em> samples inferred using the approach implemented in <code>GenomScope 2.0</code> <span class="citation">(<a href="#ref-RanalloBenavidez2020" role="doc-biblioref">Ranallo-Benavidez et al., 2020</a>)</span>, which relies on k-mers (k=21) inferred by <code>JellyFish</code> <span class="citation">(<a href="#ref-Jellyfish2011" role="doc-biblioref">Marçais and Kingsford, 2011</a>)</span>.</p>
<div id="data-requirements-2" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> Data requirements</h2>
<ul>
<li><strong>Input:</strong> <code>$SRA_R1_trimmed.fastq.gz</code> and <code>$SRA_R2_trimmed.fastq.gz</code> containing cleaned paired reads from <code>Trimmomatic</code>.</li>
<li><strong>Output:</strong> <code>21mer_trimmed_"$SRA"_pe12.histo</code> files from <code>JellyFish</code> used as input for the <code>GenomeScope 2</code> analysis. The latter analysis will generate plots stored in <code>png</code> format. You will also need to store <code>kcov</code> provided by <code>GenomeScope 2</code> for the <code>smudgeplot</code> analyses.</li>
</ul>
</div>
<div id="dependencies-3" class="section level2" number="8.2">
<h2><span class="header-section-number">8.2</span> Dependencies</h2>
<p><code>GenomScope 2.0</code> is available on <a href="https://github.com/tbenavi1/genomescope2.0">GitHub</a> and <code>JellyFish</code> is available <a href="https://genome.umd.edu/jellyfish.html#Release">here</a>.</p>
</div>
<div id="code-3" class="section level2" number="8.3">
<h2><span class="header-section-number">8.3</span> Code</h2>
<p>The commands to conduct these analyses are as follows:</p>
<pre class="bash"><code># 1. Navigate to right location
cd ../JellyFish &amp;&amp;

# 2. Generate histo file with JellyFish
jellyfish count -t 30 -C -m 21 -s 5G -o 21mer_trimmed_&quot;$SRA&quot;_pe12 &lt;(zcat /media/SeaGate/Vanilla_SRA/Trimmomatic/&quot;$SRA&quot;_1_trimmed.fastq.gz) &lt;(zcat /media/SeaGate/Vanilla_SRA/Trimmomatic/&quot;$SRA&quot;_2_trimmed.fastq.gz) &amp;&amp; 
jellyfish histo -o 21mer_trimmed_&quot;$SRA&quot;_pe12.histo 21mer_trimmed_&quot;$SRA&quot;_pe12

# 3. Use GenomeScope 2.0 to infer genome size and level of heterozygosity
# Run locally using GenomeScope R package (must be installed)
genomescope.R -i 21mer_trimmed_&quot;$SRA&quot;_pe12.histo -o output -k 21
# Or go online ti run analysis with histo file
http://qb.cshl.edu/genomescope/genomescope2.0/</code></pre>
<p><strong>Note:</strong> You need to record <code>kcov</code> from the <code>GenomeScope</code> output to set the <code>L</code> parameter in the <code>smudgeplot</code> analysis (see below).</p>
</div>
</div>
<div id="ploidy-level-and-genome-complexity" class="section level1" number="9">
<h1><span class="header-section-number">9</span> Ploidy level and genome complexity</h1>
<p>Ploidy levels of <em>Vanilla planifolia</em> samples and their genomic complexity (focusing on heterozygous k-mers) were inferred by applying the approach implemented in <code>smudgeplot</code> <span class="citation">(<a href="#ref-RanalloBenavidez2020" role="doc-biblioref">Ranallo-Benavidez et al., 2020</a>)</span>, which relies on k-mers (k=21) inferred by <code>KMC</code> <span class="citation">(<a href="#ref-KMC2017" role="doc-biblioref">Kokot et al., 2017</a>)</span>.</p>
<div id="data-requirements-3" class="section level2" number="9.1">
<h2><span class="header-section-number">9.1</span> Data requirements</h2>
<ul>
<li><strong>Input:</strong> <code>$SRA_R1_trimmed.fastq.gz</code> and <code>$SRA_R2_trimmed.fastq.gz</code> containing cleaned paired reads from <code>Trimmomatic</code> and <code>kcov</code> values from the <code>GenomeScope</code> analyses.</li>
<li><strong>Output:</strong> <code>kmcdb_"$SRA"_L"$L"_U"$U"_coverages.tsv</code> files from <code>KMC</code>. These files will be used as input for the <code>smudgeplot</code> analyses, which will output plots in <code>png</code> format.</li>
</ul>
</div>
<div id="dependencies-4" class="section level2" number="9.2">
<h2><span class="header-section-number">9.2</span> Dependencies</h2>
<p><code>KMC</code> is available on <a href="https://github.com/refresh-bio/KMC">GitHub</a> and <code>smudgeplot</code> is also available on <a href="https://github.com/KamilSJaron/smudgeplot">GitHub</a>.</p>
</div>
<div id="code-4" class="section level2" number="9.3">
<h2><span class="header-section-number">9.3</span> Code</h2>
<p>The commands to conduct these analyses are as follows:</p>
<pre class="bash"><code># 1.Navigate to right folder
cd ../Smudgeplot/ &amp;&amp;

# 2. Conduct smudgeplot to infer ploidy and genome complexity
# list files for kmc tools analysis
ls /media/SeaGate/Vanilla_SRA/Trimmomatic/&quot;$SRA&quot;_R*_trimmed.fq.gz &gt; FILES_&quot;$SRA&quot; &amp;&amp;
# kmer 21, 30 threads, 128G of memory, counting kmer coverages between 1 and 10000x
kmc -k21 -t30 -m128 -ci1 -cs10000 @FILES_&quot;$SRA&quot; kmcdb_&quot;$SRA&quot; . &amp;&amp;
kmc_tools transform kmcdb_&quot;$SRA&quot; histogram kmcdb_k21_&quot;$SRA&quot;.hist -cx10000 &amp;&amp;

# Declare variable 
kcov=32 #Please edit based on GenomeScope analysis

# Define L = (kcov/2) - 5 # As proposed in smudgeplot manual
L=$(expr $kcov / 2 - 5) &amp;&amp;
U=$(smudgeplot.py cutoff kmcdb_k21_&quot;$SRA&quot;.hist U) &amp;&amp;

echo $L $U # these need to be sane values &amp;&amp;
# L should be like 20 - 200
# U should be like 500 - 3000

kmc_tools transform kmcdb_&quot;$SRA&quot; -ci&quot;$L&quot; -cx&quot;$U&quot; reduce kmcdb_&quot;$SRA&quot;_L&quot;$L&quot;_U&quot;$U&quot; &amp;&amp;
smudge_pairs kmcdb_&quot;$SRA&quot;_L&quot;$L&quot;_U&quot;$U&quot; kmcdb_&quot;$SRA&quot;_L&quot;$L&quot;_U&quot;$U&quot;_coverages.tsv kmcdb_&quot;$SRA&quot;_L&quot;$L&quot;_U&quot;$U&quot;_pairs.tsv &gt; kmcdb_&quot;$SRA&quot;_L&quot;$L&quot;_U&quot;$U&quot;_familysizes.tsv

# Generate the smudgeplot
#Run R code to produce smudgeplot (R package must be installed)
smudgeplot.py plot kmcdb_&quot;$SRA&quot;_L&quot;$L&quot;_U&quot;$U&quot;_coverages.tsv -o &quot;$SRA&quot;_L&quot;$L&quot;_U&quot;$U&quot; -k 21</code></pre>
</div>
</div>
<div id="reconstructing-genome-sequences" class="section level1" number="10">
<h1><span class="header-section-number">10</span> Reconstructing genome sequences</h1>
<p>Genome reconstructions (at chromosome scale) were carried out by mapping the cleaned trimmed reads to the <em>Vanilla planifolia</em> Daphna reference genome (<a href="https://www.ncbi.nlm.nih.gov/data-hub/genome/GCA_016413895.1/">GCA_016413895.1</a> in <code>fasta</code> format. Read mapping was conducted with <code>bowtie2</code> <span class="citation">(<a href="#ref-Langmead2012" role="doc-biblioref">Langmead and Salzberg, 2012</a>)</span> and the consensus genome sequences (in <code>fasta</code> format) were reconstructed by using <code>samtools</code> and <code>bcftools</code> <span class="citation">(<a href="#ref-Danecek2021" role="doc-biblioref">Danecek et al., 2021</a>)</span>.</p>
<div id="data-requirements-4" class="section level2" number="10.1">
<h2><span class="header-section-number">10.1</span> Data requirements</h2>
<ul>
<li><strong>Input:</strong> <code>GCA_016413895.1_Elo_Vpla-A_principal_1.0_genomic_chrmosomes.fna</code> reference genome and <code>$SRA_R1_trimmed.fastq.gz</code> and <code>$SRA_R2_trimmed.fastq.gz</code> files.</li>
<li><strong>Output:</strong> <code>"$SRA"_consensus.fa</code> containing newly assembled genome sequences for each sample.</li>
</ul>
</div>
<div id="dependencies-5" class="section level2" number="10.2">
<h2><span class="header-section-number">10.2</span> Dependencies</h2>
<p><code>bowtie2</code> is available on <a href="https://github.com/BenLangmead/bowtie2">GitHub</a>, but it can also be installed with <a href="https://anaconda.org/">Bioconda</a> as follows:</p>
<pre class="bash"><code>#Install bowtie2 with Bioconda
conda install bowtie2</code></pre>
<p><code>samtools</code> and <code>bcftools</code> are available on <a href="https://github.com/samtools/">GitHub</a>.</p>
</div>
<div id="code-5" class="section level2" number="10.3">
<h2><span class="header-section-number">10.3</span> Code</h2>
<p>The code to execute these analyses is presented below:</p>
<pre class="bash"><code># 1. Declare variable with sample name (if not already done)
SRA=Van65 &amp;&amp; # Please adjust to your sample

# 2. Navigate to right folder
cd ../Bowtie &amp;&amp;

# 3. Create index of reference genome (this needs to be done only once)
bowtie2-build GCA_016413895.1_Elo_Vpla-A_principal_1.0_genomic_chrmosomes.fna GCA_016413895

# 4. Map reads on ref genome using bowtie
# Here we use the sensitive mode to be precise and 24 CPUs
bowtie2 --end-to-end --sensitive -p 24 -x ../Reference_genome/GCA_016413895 -1 ../Trimmomatic/&quot;$SRA&quot;_R1_trimmed.fq.gz -2 ../Trimmomatic/&quot;$SRA&quot;_R2_trimmed.fq.gz -S &quot;$SRA&quot;.sam &amp;&amp; # Not printing anything in console

# 5. Sort sam file and generate bam file
cat &quot;$SRA&quot;.sam | samtools view -bS - | samtools sort -o &quot;$SRA&quot;.bam &amp;&amp;

# 6. Call variants
bcftools mpileup  -Ou -f ../Reference_genome/GCA_016413895.1_Elo_Vpla-A_principal_1.0_genomic_chrmosomes.fna &quot;$SRA&quot;.bam | bcftools call -vmO z -o &quot;$SRA&quot;_calls.vcf.gz &amp;&amp;

# 7. Index the calls
bcftools index &quot;$SRA&quot;_calls.vcf.gz &amp;&amp;

# 8. Normalize indels
bcftools norm -f ../Reference_genome/GCA_016413895.1_Elo_Vpla-A_principal_1.0_genomic_chrmosomes.fna &quot;$SRA&quot;_calls.vcf.gz -Ob -o &quot;$SRA&quot;_calls.norm.bcf &amp;&amp;

# 9. Filter adjacent indels within 5bp
bcftools filter --IndelGap 5 &quot;$SRA&quot;_calls.norm.bcf -Ob -o &quot;$SRA&quot;_calls.norm.flt-indels.bcf &amp;&amp;

# 10. Apply variants to create consensus sequence
cat ../Reference_genome/GCA_016413895.1_Elo_Vpla-A_principal_1.0_genomic_chrmosomes.fna | bcftools consensus &quot;$SRA&quot;_calls.vcf.gz &gt; &quot;$SRA&quot;_consensus.fa</code></pre>
<p>Because scaffolds in the newly produced consensus genome sequences have to the same names than in the reference file, we have developed an <code>R</code> code (relying on base <code>R</code> functions) to rename scaffolds based on sample ID. The code is as follows:</p>
<pre class="r"><code># Rename scaffolds in consensus fasta file

# 1. Launch R from the terminal
R

# 2. Set file name
SRA &lt;- &quot;Van65&quot; # Please adjust to your sample

# 3. Load consensus fasta file
conFasta &lt;- readLines(paste0(SRA, &quot;_consensus.fa&quot;)) 

# 4. Extract old names
FastaNames &lt;- conFasta[grep(&quot;&gt;&quot;, conFasta)]

# 5. Build new names
NewNames &lt;- gsub(&quot;&gt;&quot;, paste0(&quot;&gt;&quot;, SRA, &quot;_&quot;), sapply(strsplit(FastaNames, &quot; &quot;), &quot;[[&quot;, 1))

# 6. Replace old scaffold names by new names
conFasta[grep(&quot;&gt;&quot;, conFasta)] &lt;- NewNames

# 7. Overwrite consensus file with new scaffold names
write.table(conFasta, paste0(SRA, &quot;_consensus.fa&quot;), row.names = F, col.names=F, quote = F)

# 8. Quit R
quit(save=&quot;no&quot;)</code></pre>
</div>
</div>
<div id="comparative-chromosome-level-analyses" class="section level1" number="11">
<h1><span class="header-section-number">11</span> Comparative chromosome level analyses</h1>
<p>We applied the same approach than <span class="citation"><a href="#ref-Shirasawa2022" role="doc-biblioref">Shirasawa et al.</a> (<a href="#ref-Shirasawa2022" role="doc-biblioref">2022</a>)</span> and compared chromosome-level genome sequences of <em>Vanilla planifolia</em> samples with <code>Minimap2</code> <span class="citation">(<a href="#ref-Li2018" role="doc-biblioref">Li, 2018</a>)</span>, and produced coverage maps with the R package <a href="https://dwinter.github.io/pafr/index.html"><code>pafr</code></a>. Coverage maps allowed assessing for any chromosomal re-arrangements as well as evaluating coverage of our target sample compared to the reference genome. To further investigate putative chromosomal rearrangements, we inferred dot plots with the R package <a href="https://github.com/tpoorten/dotPlotly"><code>pafCoordsDotPlotly</code></a>. Finally, a heatmap showing percentage of identities between each sample and the reference genome at chromsome level was inferred using a custom <code>R</code> script. All these comparative analyses rely on the output of the <code>Minimap2</code> analyses, which are in <a href="https://cran.r-project.org/web/packages/pafr/vignettes/Introduction_to_pafr.html#:~:text=PAF%20is%20a%20plain%20text,least%20the%20following%2012%20columns.&amp;text=In%20addition%2C%20each%20row%20can,the%20SAM%20specification%20(pdf)"><code>paf</code></a> format. The code to execute these analyses is presented below (if you need to install software see below):</p>
<div id="code-for-full-genome-alignments" class="section level2" number="11.1">
<h2><span class="header-section-number">11.1</span> Code for full genome alignments</h2>
<pre class="bash"><code>#Note: This code produces an output in paf format. 

# 1. Declare variable with sample name (if not already done)
SRA=Van69

# 2. Navigate to right folder
cd ../Minimap2 &amp;&amp;

# 3. Conduct analysis with 1% of divergence between genomes (= asm10)
minimap2 -t26 -cx asm10 -k 21 ../Reference_genome/GCA_016413895.1_Elo_Vpla-A_principal_1.0_genomic_chrmosomes.fna ../Bowtie/&quot;$SRA&quot;_consensus.fa &gt; aln_ref_asm10_&quot;$SRA&quot;.paf #  asm5/asm10/asm20 - asm-to-ref mapping, for ~0.1/1/5% sequence divergence</code></pre>
</div>
<div id="code-for-dot-plots" class="section level2" number="11.2">
<h2><span class="header-section-number">11.2</span> Code for dot plots</h2>
<pre class="r"><code># Note: This code outputs png and html documents

# 1. Navigate to right folder
cd ../DotPlot &amp;&amp;
  
# 2. Run analysis directly from terminal using Rscript
# You have to adjust file names 
Rscript pafCoordsDotPlotly.R -i /media/SeaGate/Vanilla_SRA/Minimap2/aln_ref_asm10_Van65.paf -o dotPlot_aln_ref_asm10_Van65 -s -t -m 10000 -q 50000 -k 14 -l # Provide name (change to VanXX)</code></pre>
</div>
<div id="code-for-genome-coverage-maps" class="section level2" number="11.3">
<h2><span class="header-section-number">11.3</span> Code for genome coverage maps</h2>
<pre class="r"><code># 1. Navigate to right folder
cd ../Pafr &amp;&amp;

# 2. Lauch R in terminal
R

# 3. Load package
# Warning: Please install pafr with devtools prior to executing this code (see below)
library(pafr, quietly=TRUE)

# 4. Set file name
SRA &lt;- &quot;Van65&quot; # Adjust name

# 5. Read in the PAF data (output of Minimap2)
ali &lt;- pafr::read_paf(paste0(&quot;../Minimap2/aln_ref_asm10_&quot;, SRA, &quot;.paf&quot;))

# 6. Declare a vector of colors for coverage plot
my.cols &lt;- c(&quot;orange&quot;, &quot;blue&quot;, &quot;yellow&quot;, &quot;grey&quot;, &quot;purple&quot;, &quot;green&quot;, &quot;brown&quot;, &quot;lightblue&quot;, &quot;maroon3&quot;, &quot;greenyellow&quot;,  &quot;cornsilk&quot;, &quot;coral4&quot;, &quot;chocolate&quot;, &quot;burlywood&quot;)

# 7. Infer the coverage plot and save in pdf
pdf(paste0(&quot;Coverage_plot_ref_&quot;, SRA, &quot;_asm10_Minimap2.pdf&quot;))
plot_coverage(ali, fill=&#39;qname&#39;) + scale_fill_manual(values = my.cols)
dev.off()

# 8. Quit R
quit(save=&quot;no&quot;)</code></pre>
</div>
<div id="code-for-heatmap" class="section level2" number="11.4">
<h2><span class="header-section-number">11.4</span> Code for heatmap</h2>
<p>Since the previous analyses did not unveil any chromosomal rearrangements, we can produce a heatmap to compare percentage identities between chromosomes (against the reference). To do that, we need to start by inferring the percentage identities between each target chromosome and the reference as follows: <code>100*(nmatch/alen)</code> (see <a href="https://cran.r-project.org/web/packages/pafr/vignettes/Introduction_to_pafr.html#:~:text=PAF%20is%20a%20plain%20text,least%20the%20following%2012%20columns.&amp;text=In%20addition%2C%20each%20row%20can,the%20SAM%20specification%20(pdf)">here</a> for more details). The custom <code>R</code> code below (relying on base <code>R</code> functions) builds a matrix storing percentage identities for all samples, which will then be used as input to infer the heatmap.</p>
<pre class="r"><code># 1. Navigate to right folder
cd ../Minimap2 &amp;&amp;

# 2. Launch R from terminal
R

# 3. List all the PAF files in folder
PAFfiles &lt;- list.files(pattern = &quot;.paf&quot;)

# 4. Create list of reference scaffolds/chromosomes
RefChrom &lt;- read.csv(PAFfiles[2], sep= &quot;\t&quot;, header=F)
RefChrom &lt;- unique(RefChrom[,6])

# 5. Create an empty matrix to store data (percentage identities)
PercIdMat &lt;- matrix(ncol=length(PAFfiles), nrow=length(RefChrom))
rownames(PercIdMat) &lt;- RefChrom
colnames(PercIdMat) &lt;- PAFfiles

PercIdMat &lt;- data.frame(PercIdMat)

# 6. Populate matrix with Perc identities
#Set a progress bar
pb &lt;- txtProgressBar(min = 0, max = ncol(PercIdMat), style = 3)
for(i in 1:ncol(PercIdMat)){
 #Open PAF file
 print(paste(&quot;Process: &quot;, colnames(PercIdMat)[i], sep = &quot;&quot;)) 
 PAF &lt;- read.csv(colnames(PercIdMat)[i], sep= &quot;\t&quot;, header=F)
 #Infer Perc identity for each chr
 for(j in 1:nrow(PercIdMat)){
   tmp &lt;- subset(PAF, PAF[,6] == rownames(PercIdMat)[j])
  if(nrow(tmp) &gt; 1){
   #Multiple hits so do mean
   PercIdMat[j,i] &lt;- mean(100*(tmp[,10]/tmp[,11]))
  }else{
  #Infer PercId
   PercIdMat[j,i] &lt;- 100*(tmp[,10]/tmp[,11])
  } 
 }  
  #update progress bar
  setTxtProgressBar(pb, i)  
}
close(pb)

# 7. Write data out in Heatmap
# Adjust path
write.csv(PercIdMat, file = &quot;/media/SeaGate/Vanilla_SRA/Heatmap/PercIdMat.csv&quot;, quote = F)</code></pre>
<p>Now we can generate the heatmap with the <code>R</code> <a href="https://github.com/talgalili/gplots"><em>gplots</em></a> package as follows:</p>
<pre class="r"><code># 1. Load R package
# If not installed then do
# install.packages(&quot;gplots&quot;)
library(&quot;gplots&quot;)

# 2. Edit colnames and rownames for plot (from the PercIdMat object from previous code)
colnames(PercIdMat) &lt;- gsub(&quot;aln_ref_asm10_&quot;, &quot;&quot;, colnames(PercIdMat))
colnames(PercIdMat) &lt;- gsub(&quot;.paf&quot;, &quot;&quot;, colnames(PercIdMat))
#rownames(PercIdMat) &lt;- gsub(&quot;[.]1&quot;, &quot;&quot;, rownames(PercIdMat))

# 3. Produce heatmap and save it in pdf
pdf(&quot;/media/SeaGate/Vanilla_SRA/Heatmap/Heatmap_chrom_Vanilla_2.pdf&quot;)
heatmap.2(as.matrix(PercIdMat), scale = &quot;none&quot;, col = bluered(100), 
          trace = &quot;none&quot;, density.info = &quot;none&quot;, cexRow = .8)
dev.off()</code></pre>
</div>
<div id="installing-software" class="section level2" number="11.5">
<h2><span class="header-section-number">11.5</span> Installing software</h2>
<p>The protocol to install <code>minimap2</code> is described <a href="https://github.com/lh3/minimap2">here</a>. Apart from <code>pafCoordsDotPlotly.R</code> (see below), the other R packages can be installed as follows:</p>
<pre class="r"><code># 1. Install pafr with devtools
# If you don&#39;t have devtools do:
#install.packages(devtools)
devtools::install_github(&quot;dwinter/pafr&quot;)

# 2. Install gplots from CRAN
# Documentation: https://github.com/talgalili/gplots
install.packages(&#39;gplots&#39;)</code></pre>
<p>Visit this <a href="https://github.com/tpoorten/dotPlotly">webpage</a> to install the dependencies for the <code>pafCoordsDotPlotly.R</code> script.</p>
</div>
</div>
<div id="references" class="section level1" number="12">
<h1><span class="header-section-number">12</span> References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Bolger2014" class="csl-entry">
<span class="smallcaps">Bolger, A.M.</span>, <span class="smallcaps">M. Lohse</span>, and <span class="smallcaps">B. Usadel</span>. 2014. Trimmomatic: A flexible trimmer for illumina sequence data. <em>Bioinformatics</em> 30: 2114–2120. Available at: <a href="http://dx.doi.org/10.1093/bioinformatics/btu170">http://dx.doi.org/10.1093/bioinformatics/btu170</a>.
</div>
<div id="ref-Danecek2021" class="csl-entry">
<span class="smallcaps">Danecek, P.</span>, <span class="smallcaps">J.K. Bonfield</span>, <span class="smallcaps">J. Liddle</span>, <span class="smallcaps">J. Marshall</span>, <span class="smallcaps">V. Ohan</span>, <span class="smallcaps">M.O. Pollard</span>, <span class="smallcaps">A. Whitwham</span>, et al. 2021. <span class="nocase">Twelve years of SAMtools and BCFtools</span>. <em>GigaScience</em> 10: Available at: <a href="https://doi.org/10.1093/gigascience/giab008">https://doi.org/10.1093/gigascience/giab008</a>.
</div>
<div id="ref-Hasing2020" class="csl-entry">
<span class="smallcaps">Hasing, T.</span>, <span class="smallcaps">H. Tang</span>, <span class="smallcaps">M. Brym</span>, <span class="smallcaps">F. Khazi</span>, <span class="smallcaps">T. Huang</span>, and <span class="smallcaps">A.H. Chambers</span>. 2020. A phased vanilla planifolia genome enables genetic improvement of flavour and production. <em>Nature Food</em> 1: 811–819. Available at: <a href="https://doi.org/10.1038/s43016-020-00197-2">https://doi.org/10.1038/s43016-020-00197-2</a>.
</div>
<div id="ref-KMC2017" class="csl-entry">
<span class="smallcaps">Kokot, M.</span>, <span class="smallcaps">M. Długosz</span>, and <span class="smallcaps">S. Deorowicz</span>. 2017. <span class="nocase">KMC 3: counting and manipulating k-mer statistics</span>. <em>Bioinformatics</em> 33: 2759–2761. Available at: <a href="https://doi.org/10.1093/bioinformatics/btx304">https://doi.org/10.1093/bioinformatics/btx304</a>.
</div>
<div id="ref-Langmead2012" class="csl-entry">
<span class="smallcaps">Langmead, B.</span>, and <span class="smallcaps">S.L. Salzberg</span>. 2012. Fast gapped-read alignment with bowtie 2. <em>Nature Methods</em> 9: 357–359. Available at: <a href="https://doi.org/10.1038/nmeth.1923">https://doi.org/10.1038/nmeth.1923</a>.
</div>
<div id="ref-Li2018" class="csl-entry">
<span class="smallcaps">Li, H.</span> 2018. <span class="nocase">Minimap2: pairwise alignment for nucleotide sequences</span>. <em>Bioinformatics</em> 34: 3094–3100. Available at: <a href="https://doi.org/10.1093/bioinformatics/bty191">https://doi.org/10.1093/bioinformatics/bty191</a>.
</div>
<div id="ref-Jellyfish2011" class="csl-entry">
<span class="smallcaps">Marçais, G.</span>, and <span class="smallcaps">C. Kingsford</span>. 2011. A fast, lock-free approach for efficient parallel counting of occurrences of k-mers. <em>Bioinformatics</em> 27: 764–770. Available at: <a href="+ http://dx.doi.org/10.1093/bioinformatics/btr011">+ http://dx.doi.org/10.1093/bioinformatics/btr011</a>.
</div>
<div id="ref-RanalloBenavidez2020" class="csl-entry">
<span class="smallcaps">Ranallo-Benavidez, T.R.</span>, <span class="smallcaps">K.S. Jaron</span>, and <span class="smallcaps">M.C. Schatz</span>. 2020. GenomeScope 2.0 and smudgeplot for reference-free profiling of polyploid genomes. <em>Nature Communications</em> 11: 1432. Available at: <a href="https://doi.org/10.1038/s41467-020-14998-3">https://doi.org/10.1038/s41467-020-14998-3</a>.
</div>
<div id="ref-Shirasawa2022" class="csl-entry">
<span class="smallcaps">Shirasawa, K.</span>, <span class="smallcaps">S. Ueta</span>, <span class="smallcaps">K. Murakami</span>, <span class="smallcaps">M. Abdelrahman</span>, <span class="smallcaps">A. Kanno</span>, and <span class="smallcaps">S. Isobe</span>. 2022. <span class="nocase">Chromosome-scale haplotype-phased genome assemblies of the male and female lines of wild asparagus (Asparagus kiusianus), a dioecious plant species</span>. <em>DNA Research</em> 29: Available at: <a href="https://doi.org/10.1093/dnares/dsac002">https://doi.org/10.1093/dnares/dsac002</a>.
</div>
</div>
</div>
<div id="appendix-1" class="section level1 appendix" number="13">
<h1><span class="header-section-number">13</span> Appendix 1</h1>
<p>Citations of all R packages used to generate this report.</p>
<p>[1] J. Allaire, Y. Xie, J. McPherson, et al. <em>rmarkdown: Dynamic
Documents for R</em>. R package version 2.14. 2022. &lt;URL:
<a href="https://CRAN.R-project.org/package=rmarkdown" class="uri">https://CRAN.R-project.org/package=rmarkdown</a>&gt;.</p>
<p>[2] C. Boettiger. <em>knitcitations: Citations for Knitr Markdown Files</em>.
R package version 1.0.12. 2021. &lt;URL:
<a href="https://github.com/cboettig/knitcitations" class="uri">https://github.com/cboettig/knitcitations</a>&gt;.</p>
<p>[3] M. C. Koohafkan. <em>kfigr: Integrated Code Chunk Anchoring and
Referencing for R Markdown Documents</em>. R package version 1.2.1. 2021.
&lt;URL: <a href="https://github.com/mkoohafkan/kfigr" class="uri">https://github.com/mkoohafkan/kfigr</a>&gt;.</p>
<p>[4] R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2019. &lt;URL: <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>&gt;.</p>
<p>[5] H. Wickham, J. Bryan, and M. Barrett. <em>usethis: Automate Package
and Project Setup</em>. R package version 2.1.5. 2021. &lt;URL:
<a href="https://CRAN.R-project.org/package=usethis" class="uri">https://CRAN.R-project.org/package=usethis</a>&gt;.</p>
<p>[6] H. Wickham, R. François, L. Henry, et al. <em>dplyr: A Grammar of Data
Manipulation</em>. R package version 1.0.9. 2022. &lt;URL:
<a href="https://CRAN.R-project.org/package=dplyr" class="uri">https://CRAN.R-project.org/package=dplyr</a>&gt;.</p>
<p>[7] H. Wickham, J. Hester, W. Chang, et al. <em>devtools: Tools to Make
Developing R Packages Easier</em>. R package version 2.4.3. 2021. &lt;URL:
<a href="https://CRAN.R-project.org/package=devtools" class="uri">https://CRAN.R-project.org/package=devtools</a>&gt;.</p>
<p>[8] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. ISBN 978-1138700109. Boca Raton, Florida: Chapman and
Hall/CRC, 2016. &lt;URL: <a href="https://bookdown.org/yihui/bookdown" class="uri">https://bookdown.org/yihui/bookdown</a>&gt;.</p>
<p>[9] Y. Xie. <em>bookdown: Authoring Books and Technical Documents with R
Markdown</em>. R package version 0.26. 2022. &lt;URL:
<a href="https://CRAN.R-project.org/package=bookdown" class="uri">https://CRAN.R-project.org/package=bookdown</a>&gt;.</p>
<p>[10] Y. Xie. <em>Dynamic Documents with R and knitr</em>. 2nd. ISBN
978-1498716963. Boca Raton, Florida: Chapman and Hall/CRC, 2015. &lt;URL:
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>&gt;.</p>
<p>[11] Y. Xie. <em>formatR: Format R Code Automatically</em>. R package version
1.12. 2022. &lt;URL: <a href="https://github.com/yihui/formatR" class="uri">https://github.com/yihui/formatR</a>&gt;.</p>
<p>[12] Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in
R.” In: <em>Implementing Reproducible Computational Research</em>. Ed. by V.
Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and
Hall/CRC, 2014. &lt;URL:
<a href="http://www.crcpress.com/product/isbn/9781466561595" class="uri">http://www.crcpress.com/product/isbn/9781466561595</a>&gt;.</p>
<p>[13] Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.39. 2022. &lt;URL:
<a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>&gt;.</p>
<p>[14] Y. Xie and J. Allaire. <em>tufte: Tufte’s Styles for R Markdown
Documents</em>. R package version 0.12. 2022. &lt;URL:
<a href="https://github.com/rstudio/tufte" class="uri">https://github.com/rstudio/tufte</a>&gt;.</p>
<p>[15] Y. Xie, J. Allaire, and G. Grolemund. <em>R Markdown: The Definitive
Guide</em>. ISBN 9781138359338. Boca Raton, Florida: Chapman and Hall/CRC,
2018. &lt;URL: <a href="https://bookdown.org/yihui/rmarkdown" class="uri">https://bookdown.org/yihui/rmarkdown</a>&gt;.</p>
<p>[16] Y. Xie, C. Dervieux, and E. Riederer. <em>R Markdown Cookbook</em>. ISBN
9780367563837. Boca Raton, Florida: Chapman and Hall/CRC, 2020. &lt;URL:
<a href="https://bookdown.org/yihui/rmarkdown-cookbook" class="uri">https://bookdown.org/yihui/rmarkdown-cookbook</a>&gt;.</p>
<p>[17] H. Zhu. <em>kableExtra: Construct Complex Table with kable and Pipe
Syntax</em>. R package version 1.3.4. 2021. &lt;URL:
<a href="https://CRAN.R-project.org/package=kableExtra" class="uri">https://CRAN.R-project.org/package=kableExtra</a>&gt;.</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
